File: Programmers/Practice/LV3/42884.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes){
    int answer = 0;
    sort(routes.begin(), routes.end());
    int camera = -987654321;
    for(int i = 0; i < routes.size(); i++){
        if(camera < routes[i][0]){
            answer++;
            camera = routes[i][1];
        }
        if(camera >= routes[i][1]) camera = routes[i][1];
    }
    return answer;
}

int main(){
    cout << solution({{-20,-15}, {-14,-5}, {-18,-13}, {-5,-3}}) << '\n';
}

File: Programmers/codingTest_Kit/Greedy/42884.java
import java.util.*;

class Solution {
    public int solution(int[][] routes) {
        int answer = 0;

        Arrays.sort(routes, (a, b) -> a[0] - b[0]);

        int camera = -98765432;
        
        for(int i = 0; i < routes.length; i++) {
            if(camera < routes[i][0]) {
                answer++;
                camera = routes[i][1];
            }

            if(camera >= routes[i][1]) {
                camera = routes[i][1];
            }
        }


        return answer;
    }
}